Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=8317e0b0c2234f5f1f5d54804e4093d11bc0dffa
Commit:     8317e0b0c2234f5f1f5d54804e4093d11bc0dffa
Parent:     c15c5060fc32d7de7cde76aa61e98bae1334d82e
Author:     Takashi Iwai <tiwai@suse.de>
AuthorDate: Wed Jan 14 07:56:51 2009 +0100
Committer:  Takashi Iwai <tiwai@suse.de>
CommitDate: Wed Jan 14 08:00:49 2009 +0100

    ALSA: hda - Don't reset HP pinctl in patch_sigmatel.c
    
    Resetting HP pinctl at the unplugged state may cause a sort of regression
    on some devices because of their wrong pin configuration.
    
    A simple workaround is to disable the pin reset.  This is ugly and may be
    not good from the power-saving POV (if any), but damn simple.
    
    Signed-off-by: Takashi Iwai <tiwai@suse.de>
    Cc: stable@kernel.org

 sound/pci/hda/patch_sigmatel.c |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/sound/pci/hda/patch_sigmatel.c b/sound/pci/hda/patch_sigmatel.c
index 77fcc31..103ac6d 100644
--- a/sound/pci/hda/patch_sigmatel.c
+++ b/sound/pci/hda/patch_sigmatel.c
@@ -4225,8 +4225,19 @@ static void stac92xx_hp_detect(struct hda_codec *codec)
 			continue;
 		if (presence)
 			stac92xx_set_pinctl(codec, cfg->hp_pins[i], val);
+#if 0 /* FIXME */
+/* Resetting the pinctl like below may lead to (a sort of) regressions
+ * on some devices since they use the HP pin actually for line/speaker
+ * outs although the default pin config shows a different pin (that is
+ * wrong and useless).
+ *
+ * So, it's basically a problem of default pin configs, likely a BIOS issue.
+ * But, disabling the code below just works around it, and I'm too tired of
+ * bug reports with such devices... 
+ */
 		else
 			stac92xx_reset_pinctl(codec, cfg->hp_pins[i], val);
+#endif /* FIXME */
 	}
 } 
 
